### PR TITLE
added knative e2e tests

### PR DESCRIPTION
**Pull request by liadlauber**

### CODE REVIEW

File: test/k8s_tests/knative_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+		}, TimeoutCapp, CappCreationInterval).Should(Equal("memory"))
+	})
+
+	It("Should update ksvc container name and create a new revision when updating a capp container name", func() {
+		By("Creating a capp instance")
+		testCapp := mock.CreateBaseCapp()
+		toBeUpdatedCapp := createAndGetCapp(testCapp)
+
+		By("Updating the a capp container name")
+		newContainerName := utilst.GenerateCappName()
```
Comment (fran2810) [inline test/k8s_tests/knative_e2e_test.go:158]: generate capp name and you creating container ? you can expose a different function for this. for generating random name

File: test/k8s_tests/knative_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+	cappServiceAccount      = "system:serviceaccount:capp-operator-system:capp-operator-controller-manager"
+	exampleDanaAnnotation   = "rcs.dana.io/app-name"
+	testContainerName       = "capp-test-container"
+)
+
+// updateCapp updates the given Capp object and ensures the readiness of the latest revision
+// if shouldRevisionBeReady is true. It also checks and asserts the state of the LatestReadyRevision.
+func updateCapp(capp *rcsv1alpha1.Capp, shouldRevisionBeReady bool) {
+	latestReadyRevisionBeforeUpdate := capp.Status.KnativeObjectStatus.ConfigurationStatusFields.LatestReadyRevisionName
+	nextRevisionName := utilst.GetNextRevisionName(latestReadyRevisionBeforeUpdate)
```
Comment (fran2810) [inline test/k8s_tests/knative_e2e_test.go:29]: shouldn't it be only in the shouldbeready ?

Comment (liadlauber) [reply test/k8s_tests/knative_e2e_test.go:30]: No, I still have to check that the revision is being created even if it shouldn't be ready

**Review by fran2810 — APPROVED** [review]
Nice work!

**Pull request merged**